src/gear.py

```diff
@@ -285,3 +285,76 @@ def on_attack(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effec
         return no_effect
 
 # TODO: doppelganger brooch
+
+
+@gear
+class SenseiRobe(Gear):
+    @staticmethod
+    def on_attack(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effect:
+        if attack['type'] == Types.melee and attack['class'] == 'Special':
+            return Effect(damage=1.25)
+        return no_effect
+
+
+@gear
+class AloeVera(Gear):
+    @staticmethod
+    def on_attack(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effect:
+        if target.types[0] == Types.toxic or target.types[1] == Types.toxic:
+            return Effect(damage=1.15)
+        return no_effect
+
+
+@gear
+class TinfoilHat(Gear):
+    @staticmethod
+    def on_hit(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effect:
+        if attack['type'] == Types.digital:
+            return Effect(damage=0.7)
+        return no_effect
+
+
+@gear
+class Taser(Gear):
+    @staticmethod
+    def on_attack(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effect:
+        if attack['type'] == Types.electric and attack['class'] == 'Special':
+            return Effect(target={Statuses.burned: 1})
+        return no_effect
+
+
+@gear
+class Matcha(Gear):
+    @staticmethod
+    def on_rest(target) -> Effect:
+        return Effect(target={Statuses.cold: 0, Stats.Sta: target.max_sta * 0.4})
+
+
+@gear
+class ReactiveVial(Gear):
+    @staticmethod
+    def after_hit(attacker: TemTem, target: TemTem, attack: Dict[str, Any]) -> Effect:
+        if effectiveness(attack['type'], target) >= 1:
+            return Effect(target={Stats.HP: target.max_hp * 0.15, Statuses.nullified: 1, 'remove gear': True})
```

Comment:
The linter complains about this line, splitting it into a few more lines would probably make it more readable